proto/asn1-pdu/x509_certificate.proto

```diff
@@ -134,13 +134,36 @@ message UniqueIdentifier {
   required asn1_universal_types.BitString value = 2;
 }
 
+message Extension {
```

Comment:
Separately, I think we want to split this into two types, to follow the pattern and rubric we said for conversions

```protobuf
message Extensions {
  optional asn1_pdu.PDU pdu = 1;
  required ExtensionSequence value = 2;
}

message ExtensionSequence {
  // From RFC 5280, 4.1
  // Extensions  ::=  SEQUENCE SIZE (1..MAX) OF Extension
  // This ensures that, when not using the raw PDU for Extensions,
  // that there is at least one Extension present.
  required Extension extension = 1;
  repeated Extension extensions = 2;
}
```

This makes sure that the overall `Extensions` field can be an arbitrary PDU, and when it's structured, it's structured as valid DER that ensures that there is at least one extension.

Whether or not the `ExtensionsSequence` approach above makes sense depends on whether or not we provide a guarantee that `Extension` will always result in an encoding, which is discussed a little in the other comment and discussed in our 1:1 yesterday.